pkg/destroy/azure/azure.go

```diff
@@ -467,25 +467,24 @@ func deleteResourceGroup(ctx context.Context, client resources.GroupsClient, log
 	ctx, cancel := context.WithTimeout(ctx, 30*time.Minute)
 	defer cancel()
 
-	delFuture, err := client.Delete(ctx, name)
+	err := deleteAndWait(ctx, client, name)
 	if err != nil {
-		if wasNotFound(delFuture.Response()) {
-			logger.Debug("already deleted")
-			return nil
+		if detailedErr, ok := err.(autorest.DetailedError); ok &&
+			(detailedErr.StatusCode == http.StatusForbidden || detailedErr.StatusCode == http.StatusNotFound) {
+			return errors.Wrapf(err, "failed to delete %s", name)
 		}
-		return err
 	}
+	logger.Info("deleted")
+	return nil
+}
 
-	err = delFuture.WaitForCompletionRef(ctx, client.Client)
+func deleteAndWait(ctx context.Context, client resources.GroupsClient, resourceGroupName string) error {
```

Comment:
> forego the function entirely

+1.

Another option, which might be more go-ish (if more verbose):

```go
delFuture, err := client.Delete(ctx, name)
if err != nil {
    return err
}
err = delFuture.WaitForCompletionRef(ctx, client.Client)
if err != nil {
    if wasNotFound(delFuture.Response()) {   // or per above comment: if isNotFoundError(err) {
        logger.Debug("already deleted")
        return nil
    }
    return err
}
```

Reply:
@2uasimojo  I think with the above option we will still have nil dereference mentioned in #5428. No?

Reply:
I don't think so.
When calling `WaitForCompletionRef` (which is also part of FutureAPI) the assumption is that `delFuture.FutureAPI` can't be `nil` if `err == nil`, which (IIUC) is a typical pattern in go.
And since that's the case, subsequently calling `Response` is safe.

Reply:
My objection to the option proposed by @2uasimojo is that it effectively reverts https://github.com/openshift/installer/pull/3416, which was at least a partial fix for a real problem. There are some scenarios where the call to `client.Delete` will return an error when the resource group does not exist.

Reply:
Oh, so we *do* have to check `err` for 404 immediately after the `Delete` call? Then yeah, what you said. Which, for completeness, is approximately the following if you accept my above comment about the helper function:
```go
delFuture, err := client.Delete(ctx, name)
if err == nil {
    err = delFuture.WaitForCompletionRef(ctx, client.Client)
}
if err != nil {
    if isNotFoundError(err) {
        logger.Debug("already deleted")
        return nil
    }
    return errors.Wrapf(err, "failed to delete %s", name)
}
logger.Info("deleted")
return nil
```

Reply:
Thanks you @2uasimojo and @staebler! I've incorporated your suggestions into the PR. PTAL. Thanks.